src/pika_rm.cc

```diff
@@ -325,6 +325,24 @@ Status SyncMasterPartition::GetLastRecvTime(const std::string& ip, int port, uin
   return Status::OK();
 }
 
+bool SyncMasterPartition::CheckSlaveBinlogConsistent(const std::string &ip, int port) {
+  std::shared_ptr<SlaveNode> slave_ptr = GetSlaveNode(ip, port);
+  if (!slave_ptr) {
+    LOG(WARNING) << "ip " + ip + " port " + std::to_string(port) << " not found";
+    return false;
+  }
+
+  if (slave_ptr->acked_offset == slave_ptr->sent_offset) {
```

Comment:
when normal binlog syncing this condition it is not always true.

Reply:
For example, we send 1000 binlog at once. It may packed into 100 pb package and send. While, sent_offset is 1000, and packed_offset is increasing with the ack coming back. During all ack coming back this condition is not true.